.circleci/config.yml

```diff
@@ -1,176 +1,149 @@
+            echo "Undocumented declarations:"
+            cat "$undocumented"
+            exit 1
+          fi
+      - run: |
+          if [ $CIRCLE_BRANCH == master ]; then
+            git config --global user.email "$GH_EMAIL"
+            git config --global user.name "$GH_NAME"
 
-          git push origin gh-pages
-        fi
+            git clone $CIRCLE_REPOSITORY_URL out
 
-  - &steps-for-spm
-    - checkout
-    - run: swift test
-    - run: swift test -Xswiftc -DUSE_UTF8
+            cd out
+            git checkout gh-pages
+            git rm -rf .
+            cd ..
 
-  - &steps-for-linux
-    - checkout
-    - run: swift test
-    - run: swift test -Xswiftc -DUSE_UTF8
-    - run: mkdir -p build/reports/
-    - run: swiftlint lint --reporter junit > build/reports/junit.xml
-    - *store_test_results
+            cp -a docs/. out/.
+            cd out
 
-  - &steps-for-linux-without-swiftlint
-    - checkout
-    - run: swift test
-    - run: swift test -Xswiftc -DUSE_UTF8
+            git add -A
+            git commit -m "Automated deployment to GitHub Pages: ${CIRCLE_SHA1}" --allow-empty
 
-version: 2
-jobs:
+            git push origin gh-pages
+          fi
 
-  xcode_9.2_swift_4.0.3:
+  swiftpm:
+    parameters:
+      xcode:
+        type: string
     macos:
-      xcode: "9.2.0"
-    << : *XCODE9
-
-  xcode_9.4_swift_4.1.2:
-    macos:
-      xcode: "9.4.1"
-    << : *XCODE9
-
-  xcode_10_swift_4.2:
-    macos:
-      xcode: "10.0.0"
-    << : *XCODE10
-
-  jazzy:
-    docker:
-      - image: norionomura/jazzy:0.9.3_swift-4.1.3
-    steps: *steps-for-jazzy
-
-  spm_swift_4.0.3:
-    macos:
-      xcode: "9.2.0"
-    steps: *steps-for-spm
-
-  spm_swift_4.1.2:
-    macos:
-      xcode: "9.4.1"
-    steps: *steps-for-spm
-
-  spm_swift_4.2:
-    macos:
-      xcode: "10.0.0"
-    steps: *steps-for-spm
-
-  linux_swift_4.0.3:
-    docker:
-      - image: norionomura/swiftlint:swift-4.0.3
-    steps: *steps-for-linux
-
-  linux_swift_4.1.3:
-    docker:
-      - image: norionomura/swiftlint:swift-4.1.3
-    steps: *steps-for-linux
+      xcode: << parameters.xcode >>
+    steps: &steps-for-swiftpm
+      - checkout
+      - run: swift test
+      - run: swift test -Xswiftc -DUSE_UTF8
 
-  linux_swift_4.2:
+  linux_on_docker:
+    parameters:
+      image:
+        type: string
     docker:
-      - image: norionomura/swift:swift-4.2-branch
-    steps: *steps-for-linux-without-swiftlint
+      - image: << parameters.image >>
+    steps: *steps-for-swiftpm
 
 workflows:
   version: 2
   workflow:
     jobs:
-      - xcode_9.2_swift_4.0.3
-      - xcode_9.4_swift_4.1.2
-      - xcode_10_swift_4.2
+      - xcode:
+          name: xcode_9.2_swift_4.0.3
+          xcode: "9.2.0"
+          post-steps: [pod_lib_lint]
+      - xcode:
+          name: xcode_9.4_swift_4.1.2
+          xcode: "9.4.1"
+          post-steps: [pod_lib_lint]
```

Comment:
If we run pod lib lint on anything, shouldn't it be the latest Xcode version?

Reply:
I think that the old version is sufficient until cocoapods supporting Xcode 10 is released.
Or should we test with beta version of cocoapods?

Reply:
Oh right, that's why we're not doing this on Xcode 10, because CocoaPods stable doesn't yet support it. Good for now then! 👍